File: VB/SchedulerControl_WPF_API/CodeExamples/LabelsAndStatuses.vb
Imports DevExpress.Xpf.Scheduler
Imports DevExpress.XtraScheduler
Imports System
Imports System.Windows.Media

Namespace SchedulerControl_WPF_API
    Friend Class LabelsAndStatuses
        Private Shared Sub CustomLabelsAndStatusesAction(ByVal scheduler As SchedulerControl)
'            #Region "#CustomLabelsAndStatuses"
            scheduler.Storage.AppointmentStorage.Clear()

            Dim IssueList() As String = { "Consultation", "Treatment", "X-Ray" }
            Dim IssueColorList() As Color = { Colors.Ivory, Colors.Pink, Colors.Plum }
            Dim PaymentStatuses() As String = { "Paid", "Unpaid" }
            Dim PaymentColorStatuses() As Color = { Colors.Green, Colors.Red }


            Dim labelStorage As IAppointmentLabelStorage = scheduler.Storage.AppointmentStorage.Labels
            labelStorage.Clear()
            Dim count As Integer = IssueList.Length
            For i As Integer = 0 To count - 1
                Dim label As IAppointmentLabel = labelStorage.CreateNewLabel(i, IssueList(i))
                label.SetColor(IssueColorList(i))
                labelStorage.Add(label)
            Next i
            Dim statusStorage As IAppointmentStatusStorage = scheduler.Storage.AppointmentStorage.Statuses
            statusStorage.Clear()
            count = PaymentStatuses.Length
            For i As Integer = 0 To count - 1
                Dim status As IAppointmentStatus = statusStorage.CreateNewStatus(i, PaymentStatuses(i), PaymentStatuses(i))
                status.SetBrush(New SolidColorBrush(PaymentColorStatuses(i)))
                statusStorage.Add(status)
            Next i

            ' Create a new appointment.
            Dim apt As Appointment = scheduler.Storage.CreateAppointment(AppointmentType.Normal)
            apt.Subject = "Test"
            apt.Start = Date.Now
            apt.End = Date.Now.AddHours(2)
            apt.ResourceId = scheduler.Storage.ResourceStorage(0).Id
            apt.LabelKey = labelStorage.GetByIndex(2).Id
            scheduler.Storage.AppointmentStorage.Add(apt)

'            #End Region ' #CustomLabelsAndStatuses
        End Sub
    End Class
End Namespace

File: CS/SchedulerControl_WPF_API/CodeExamples/LabelsAndStatuses.cs
using DevExpress.Xpf.Scheduler;
using DevExpress.XtraScheduler;
using System;
using System.Windows.Media;

namespace SchedulerControl_WPF_API {
    class LabelsAndStatuses
    {
        static void CustomLabelsAndStatusesAction(SchedulerControl scheduler)
        {
            #region #CustomLabelsAndStatuses
            scheduler.Storage.AppointmentStorage.Clear();

            string[] IssueList = { "Consultation", "Treatment", "X-Ray" };
            Color[] IssueColorList = { Colors.Ivory, Colors.Pink, Colors.Plum };
            string[] PaymentStatuses = { "Paid", "Unpaid" };
            Color[] PaymentColorStatuses = { Colors.Green, Colors.Red };


            IAppointmentLabelStorage labelStorage = scheduler.Storage.AppointmentStorage.Labels;
            labelStorage.Clear();
            int count = IssueList.Length;
            for (int i = 0; i < count; i++)
            {
                IAppointmentLabel label = labelStorage.CreateNewLabel(i, IssueList[i]);
                label.SetColor(IssueColorList[i]);
                labelStorage.Add(label);
            }
            IAppointmentStatusStorage statusStorage = scheduler.Storage.AppointmentStorage.Statuses;
            statusStorage.Clear();
            count = PaymentStatuses.Length;
            for (int i = 0; i < count; i++)
            {
                IAppointmentStatus status = statusStorage.CreateNewStatus(i, PaymentStatuses[i], PaymentStatuses[i]);
                status.SetBrush(new SolidColorBrush(PaymentColorStatuses[i]));
                statusStorage.Add(status);
            }

            // Create a new appointment.
            Appointment apt = scheduler.Storage.CreateAppointment(AppointmentType.Normal);
            apt.Subject = "Test";
            apt.Start = DateTime.Now;
            apt.End = DateTime.Now.AddHours(2);
            apt.ResourceId = scheduler.Storage.ResourceStorage[0].Id;
            apt.LabelKey = labelStorage.GetByIndex(2).Id;
            scheduler.Storage.AppointmentStorage.Add(apt);

            #endregion #CustomLabelsAndStatuses
        }
    }
}
